### ISSUE TITLE
Incorrect result when inserting into DATE column when client timezone is ahead of UTC

### ISSUE THREAD

**Issue by stegus64** [snowflakedb/snowflake-connector-net#154]
### Issue description
The test SFDbDataReader.TestGetDate() fails when running on a client that is ahead of UTC (Such as UTC+1, Stockholm)

The reason is probably the following code in SFDataConverter:208
```
                        long millis = (long)((DateTime)srcVal).ToUniversalTime().Subtract(
                            UnixEpoch).TotalMilliseconds;

```
ToUniversalTime() will change the date by subtracting from midnight when the client is ahead of UTC.
This means the wrong value is inserted into the database.

Everything works correctly when the client is after UTC such as in America.

The call to ToUniversalTime() should simply be removed.

### Configuration
*Driver version:* 1.0.12

*Dotnet framework and version:* .net standard 2.0

*Client OS:*  Windows 10

**Comment by stegus64** [comment]
@ChTimTsubasa Please remember that this issue has not been fixed. pr #156 should fix this, but that pr has been closed without merging.

**Comment by ChTimTsubasa** [comment]
Merged through #168
